lib/ret/media_search.ex

```diff
@@ -250,6 +254,52 @@ defmodule Ret.MediaSearch do
     end
   end
 
+  defp assets_search(cursor, type, account_id, query, order \\ [desc: :updated_at]) do
+    results =
+      Asset
+      |> where([a], a.account_id == ^account_id and a.type == ^type)
+      |> add_query_to_asset_search_query(query)
+      |> preload([:asset_owned_file])
+      |> order_by(^order)
+      |> Repo.paginate(%{page: cursor, page_size: @page_size})
+      |> result_for_assets_page(cursor)
+
+    {:commit, results}
+  end
+
+  defp add_query_to_asset_search_query(query, nil), do: query
+  defp add_query_to_asset_search_query(query, q), do: query |> where([a], ilike(a.name, ^"%#{q}%"))
+
+  defp result_for_assets_page(page, cursor) do
+    %Ret.MediaSearchResult{
+      meta: %Ret.MediaSearchResultMeta{
+        next_cursor:
+          if page.total_pages > cursor do
+            (cursor || 1) + 1
```

Comment:
the pagination recently changed today on master -- you'll need to update this code after pulling from master. the `cursor` being passed in now should always be a string, and be parsed in this module. (see how its done for scenes on master, and in media_search_controller for scene_listing search)